### ISSUE TITLE
run series of hypercert minting & claiming cases

### ISSUE THREAD

**Issue by ccerv1** [hypercerts-org/hypercerts#421]
### What is it?

Each test should capture the following:

- Test description
- Minting URL (for reproduction)
- Expected behavior
- Observed behavior

**Comment by ccerv1** [comment]
# Test 1: Create a hypercert with no allowlist

[url](https://testnet.hypercerts.org/app/create?name=Hypercert%20Test%201&description=Create%20a%20hypercert%20with%20no%20allowlist%0A%0AAfter%20minting%2C%20I%20expect%20the%20hypercert%20to%20appear%20in%20my%20dashboard%2C%20listed%20as%20a%20hypercert%20I%20created%20and%20that%20I%20claimed%20%28without%20performing%20the%20claim%20transaction%29.%0A%0AI%20also%20expect%20to%20see%20that%20I%20own%20100%25%20of%2010%2C000%20units.&logoUrl=https%3A%2F%2Fi.imgur.com%2FsDQhp3Y.png&bannerUrl=https%3A%2F%2Fi.imgur.com%2FwsM3fWd.jpeg&impactScopes%5B0%5D=all&impactTimeEnd=indefinite&workScopes=testing&workTimeStart=2023-01-01&workTimeEnd=2023-02-28&rights%5B0%5D=Public%20Display&contributors=0xC54861e4161E02354957A1a2F95303489CD60833&backgroundColor=blue&backgroundVectorArt=contours)

**Expected behavior**
After minting, I expect the hypercert to appear in my dashboard, listed as a hypercert I created and that I claimed (without performing the claim transaction).

When I view the hypercert, I expect to see that I own 100% of 10,000 units.

**Observed behavior**
The hypercert appeared in my dashboard stats as both created and claimed (correct), but only listed once in my collection (correct).

When I view [the hypercert](https://testnet.hypercerts.org/app/view?claimId=0x822f17a9a5eecfd66dbaff7946a8071c265d1d07-15652988878363169319315231941861337726976), I see that I own 100% of 10,000 units (correct).

**Comment by ccerv1** [comment]
# Test 2: Create a hypercert and allowlist the same wallet (mixed case)

**Minting Link**
[Here](https://testnet.hypercerts.org/app/create?name=Hypercert%20Test%202&description=Create%20a%20hypercert%20and%20allowlist%20the%20same%20wallet%20%28mixed%20case%29%0A%0AAfter%20minting%2C%20I%20expect%20the%20hypercert%20to%20appear%20in%20my%20dashboard%20with%20the%20following%20stats%3A%20%0A-%202%20created%20%0A-%201%2F2%20claimed%0A%0AHypercert%202%20should%20only%20appear%20once%20in%20the%20list%2C%20as%20a%20claimable%20hypercert%0A%0AI%20should%20be%20able%20to%20successfully%20claim%20the%20hypercert.%0A%0AThen%20I%20expect%20following%20stats%20to%20say%202%2F2%20claimed%20and%20the%20hypercert%20chip%20to%20show%20%22claimed%22%20only.%0A%0AIf%20I%20view%20the%20hypercert%2C%20I%20should%20own%20100%25%20of%20it.&logoUrl=https%3A%2F%2Fi.imgur.com%2FsDQhp3Y.png&bannerUrl=https%3A%2F%2Fi.imgur.com%2FwsM3fWd.jpeg&impactScopes%5B0%5D=all&impactTimeEnd=indefinite&workScopes=testing&workTimeStart=2023-01-01&workTimeEnd=2023-02-28&rights%5B0%5D=Public%20Display&contributors=0xC54861e4161E02354957A1a2F95303489CD60833&allowlistUrl=https%3A%2F%2Fbafkreidmtx2ilf7ys6lfvvvmrjwhjo3qeqy4cidj5z7mdclbhawc62hdhi.ipfs.nftstorage.link%2F&backgroundColor=blue&backgroundVectorArt=contours)
Allowlist [here](https://bafkreidmtx2ilf7ys6lfvvvmrjwhjo3qeqy4cidj5z7mdclbhawc62hdhi.ipfs.nftstorage.link/)

**Expected behavior**
After minting, I expect the hypercert to appear in my dashboard with the following stats: 
- 2 created 
- 1/2 claimed

Hypercert 2 should only appear once in the list, as a claimable hypercert

I should be able to successfully claim the hypercert.

Then I expect following stats to say 2/2 claimed and the hypercert chip to show "claimed" only.

If I view the hypercert, I should own 100% of it.

**Observed behavior**
After minting, I see the following (there was some latency -- at first I only saw the created entry, not the claimable one). The headline stats are correct, however, the same hypercert is listed as both created and claimable.
<img width="1248" alt="image" src="https://user-images.githubusercontent.com/42869436/222468051-3873c5c4-7b4a-481c-81d7-46a2d36ed725.png">

Claiming works. There is only one entry on the list (correct). However, my stats now display 2/3 claimed
<img width="1196" alt="image" src="https://user-images.githubusercontent.com/42869436/222468588-8c00619f-6e0b-4683-8376-1a32ea77030c.png">

If I view the hypercert, I see that I own 100% (correct)

When I come back to the dashboard, it now only shows 2/2 claimed (correct)

<img width="1141" alt="image" src="https://user-images.githubusercontent.com/42869436/222468927-1c7e631f-101e-4531-bd1e-8a2bdec6645d.png">

**Comment by ccerv1** [comment]
# Test 3: Create a hypercert and allowlist a different wallet

**Links**
- Allowlist [here](https://bafkreice2hkp3racxdxbnptgwor33aj6k6z4svunbzfeqlhheww3wa3iju.ipfs.nftstorage.link/)
- Minting [url](https://testnet.hypercerts.org/app/create?name=Hypercert%20Test%203&description=Create%20a%20hypercert%20and%20allowlist%20a%20different%20wallet%0A%0AAfter%20minting%2C%20I%20expect%20the%20hypercert%20to%20appear%20in%20my%20dashboard%20with%20the%20following%20stats%3A%20%0A-%203%20created%20%0A-%202%2F2%20claimed%0A%0AHypercert%203%20should%20only%20appear%20as%20a%20created%20hypercert.%0A%0AIf%20I%20switch%20to%20my%20other%20wallet%2C%20I%20should%20see%200%2F1%20stats%20and%20a%20claimable%20hypercert%20in%20my%20list.%0A%0AI%20should%20be%20able%20to%20successfully%20claim%20the%20hypercert.%0A%0AThen%20I%20expect%20my%20stats%20to%20say%201%2F1%20claimed%20and%20the%20hypercert%20chip%20to%20show%20%22claimed%22%20only.%0A%0AIf%20I%20view%20the%20hypercert%2C%20I%20should%20own%20100%25%20of%20it.%0A%0AIf%20I%20switch%20wallets%2C%20I%20should%20view%20that%20the%20hypercert%20is%20100%25%20owned%20by%20my%20other%20wallet%20address.&logoUrl=https%3A%2F%2Fi.imgur.com%2FsDQhp3Y.png&bannerUrl=https%3A%2F%2Fi.imgur.com%2FwsM3fWd.jpeg&impactScopes%5B0%5D=all&impactTimeEnd=indefinite&workScopes=testing&workTimeStart=2023-01-01&workTimeEnd=2023-02-28&rights%5B0%5D=Public%20Display&contributors=0xC54861e4161E02354957A1a2F95303489CD60833&allowlistUrl=https%3A%2F%2Fbafkreice2hkp3racxdxbnptgwor33aj6k6z4svunbzfeqlhheww3wa3iju.ipfs.nftstorage.link%2F&backgroundColor=blue&backgroundVectorArt=contours)

**Expected Behavior**

After minting, I expect the hypercert to appear in my dashboard with the following stats: 
- 3 created 
- 2/2 claimed

Hypercert 3 should only appear as a created hypercert.

If I switch to my other wallet, I should see 0/1 stats and a claimable hypercert in my list.

I should be able to successfully claim the hypercert.

Then I expect my stats to say 1/1 claimed and the hypercert chip to show "claimed" only.

If I view the hypercert, I should own 100% of it.

If I switch wallets, I should view that the hypercert is 100% owned by my other wallet address.

**Observed Behavior**

After minting, I see the correct stats:
- 3 created 
- 2/2 claimed

Hypercert 3 only appears as a created hypercert (correct).

If I switch to my other wallet, I see 0/1 stats and a claimable hypercert in my list (correct).

I am able to successfully claim the hypercert.

At first, my stats show 1/2 claimed (though I only see one hypercert listed).
<img width="1156" alt="image" src="https://user-images.githubusercontent.com/42869436/222478830-94fa58e3-9ea1-4b5e-a88e-8724f26d4a10.png">

If I view the hypercert, I own 100% of it (correct).

If I go back to my dashboard, my stats now say 1/1 claimed and the hypercert chips show "claimed" only (correct).

If I switch wallets, I view that the hypercert is 100% owned by my other wallet address (correct).

**Comment by ccerv1** [comment]
# Test 4: Create a hypercert and allowlist a different wallet that gets repeated (different cases)

**Links**
- Allowlist [here](https://bafkreifbxjmxyiguzxnztxeonbjwygun4aql53pi65s3dcvdck3a7laxke.ipfs.nftstorage.link/)
- Minting [url](https://testnet.hypercerts.org/app/create?name=Hypercert%20Test%204&description=Create%20a%20hypercert%20and%20allowlist%20a%20different%20wallet%20that%20gets%20repeated%20%28different%20cases%29%0A%0AAfter%20minting%2C%20I%20expect%20the%20hypercert%20to%20appear%20in%20my%20dashboard%20with%20the%20following%20stats%3A%20%0A-%204%20created%20%0A-%202%2F2%20claimed%0A%0AHypercert%204%20should%20only%20appear%20as%20a%20created%20hypercert.%0A%0AIf%20I%20switch%20to%20my%20other%20wallet%2C%20I%20should%20see%201%2F2%20stats%20and%20a%20claimable%20hypercert%20in%20my%20list.%0A%0AI%20should%20be%20able%20to%20successfully%20claim%20the%20hypercert.%0A%0AThen%20I%20expect%20my%20stats%20to%20say%202%2F2%20claimed%20and%20the%20hypercert%20chip%20to%20show%20%22claimed%22%20only.%0A%0AIf%20I%20view%20the%20hypercert%2C%20I%20should%20own%20100%25%20of%20it.%0A%0AIf%20I%20switch%20wallets%2C%20I%20should%20view%20that%20the%20hypercert%20is%20100%25%20owned%20by%20my%20other%20wallet%20address.%0A&logoUrl=https%3A%2F%2Fi.imgur.com%2FsDQhp3Y.png&bannerUrl=https%3A%2F%2Fi.imgur.com%2FwsM3fWd.jpeg&impactScopes%5B0%5D=all&impactTimeEnd=indefinite&workScopes=testing&workTimeStart=2023-01-01&workTimeEnd=2023-02-28&rights%5B0%5D=Public%20Display&contributors=0xC54861e4161E02354957A1a2F95303489CD60833&allowlistUrl=https%3A%2F%2Fbafkreifbxjmxyiguzxnztxeonbjwygun4aql53pi65s3dcvdck3a7laxke.ipfs.nftstorage.link%2F&backgroundColor=blue&backgroundVectorArt=contours)

**Expected Behavior**

After minting, I expect the hypercert to appear in my dashboard with the following stats: 
- 4 created 
- 2/2 claimed

Hypercert 4 should only appear as a created hypercert.

If I switch to my other wallet, I should see 1/2 stats and a claimable hypercert in my list.

I should be able to successfully claim the hypercert.

Then I expect my stats to say 2/2 claimed and the hypercert chip to show "claimed" only.

If I view the hypercert, I should own 100% of it.

If I switch wallets, I should view that the hypercert is 100% owned by my other wallet address.

**Observed Behavior**

After minting, I see the correct stats:
- 4 created 
- 2/2 claimed

Hypercert 4 only appears as a created hypercert (correct).

If I switch to my other wallet, I see 1/3 stats and a new claimable hypercert listed twice in my list (BUG).

<img width="1188" alt="image" src="https://user-images.githubusercontent.com/42869436/222487116-5fe92c97-40a1-437d-b51e-4e571c1bb92d.png">

I am NOT able to successfully claim the hypercert.

<img width="1293" alt="image" src="https://user-images.githubusercontent.com/42869436/222487345-3eb5bb80-afea-4fac-bf18-8599b13dd35a.png">

[stop test]

**Comment by ccerv1** [comment]
# Test 5: Create a hypercert and include the same wallet that created it in the allowlist

**Links**
- Allowlist [here](https://bafkreif2efqrxygyon2fi5rnpvqfsn2pe5ttwb6kepgksfn2jd4zka67ti.ipfs.nftstorage.link/)
- Minting [url](https://testnet.hypercerts.org/app/create?name=Hypercert%20Test%205&description=%23%20Create%20a%20hypercert%20and%20include%20the%20same%20wallet%20that%20created%20it%20in%20the%20allowlist%0A%0AAfter%20minting%2C%20I%20expect%20the%20hypercert%20to%20appear%20in%20my%20dashboard%20with%20the%20following%20stats%3A%20%0A-%206%20created%20%0A-%202%2F3%20claimed%0A%0AHypercert%205%20should%20only%20appear%20as%20a%20created%20hypercert.%0A%0AI%20should%20be%20able%20to%20claim%20it%20from%20this%20wallet%20%28the%20one%20that%20created%20it%29.%0A%0AIf%20I%20switch%20to%20my%20other%20wallet%2C%20I%20should%20see%202%2F3%20stats%20and%20a%20claimable%20hypercert%20in%20my%20list.%0A%0AI%20should%20be%20able%20to%20successfully%20claim%20the%20hypercert.%0A%0AThen%20I%20expect%20my%20stats%20to%20say%203%2F3%20claimed%20and%20the%20hypercert%20chip%20to%20show%20%22claimed%22%20only.%0A%0AIf%20I%20view%20the%20hypercert%2C%20I%20should%20own%20100%25%20of%20it.%0A%0AIf%20I%20switch%20wallets%2C%20I%20should%20view%20that%20the%20hypercert%20is%2050%25%20owned%20by%20my%20other%20wallet%20address.&logoUrl=https%3A%2F%2Fi.imgur.com%2FsDQhp3Y.png&bannerUrl=https%3A%2F%2Fi.imgur.com%2FwsM3fWd.jpeg&impactScopes%5B0%5D=all&impactTimeEnd=indefinite&workScopes=testing&workTimeStart=2023-01-01&workTimeEnd=2023-02-28&rights%5B0%5D=Public%20Display&contributors=0xC54861e4161E02354957A1a2F95303489CD60833&allowlistUrl=https%3A%2F%2Fbafkreif2efqrxygyon2fi5rnpvqfsn2pe5ttwb6kepgksfn2jd4zka67ti.ipfs.nftstorage.link%2F&backgroundColor=blue&backgroundVectorArt=contours)

**Expected Behavior**

After minting, I expect the hypercert to appear in my dashboard with the following stats: 
- 6 created 
- 2/3 claimed

Hypercert 5 should only appear as a created hypercert.

I should be able to claim it from this wallet (the one that created it).

If I switch to my other wallet, I should see 2/3 stats and a claimable hypercert in my list.

I should be able to successfully claim the hypercert.

Then I expect my stats to say 3/3 claimed and the hypercert chip to show "claimed" only.

If I view the hypercert, I should own 100% of it.

If I switch wallets, I should view that the hypercert is 50% owned by my other wallet address.

**Observed Behavior**

After minting, I see these stats:
- 6 created [correct]
- 2/2 claimed [incorrect -- should be 2/3 claimed]

I am not able to claim it from this wallet (the one that created it).

A few minutes later I came back and the stats are correct.

<img height="200" alt="image" src="https://user-images.githubusercontent.com/42869436/222508758-f1718473-ec45-4e43-855b-ef20802037a9.png">

I am able to claim the hypercert and view it in my colleciton.

<img height="200" alt="image" src="https://user-images.githubusercontent.com/42869436/222509045-30ce9a99-7e71-45eb-bb6b-71fb2cc469ce.png">

If I switch to my other wallet, I see 1/4 stats and a new claimable hypercert in my list (correct).

<img height="200" alt="image" src="https://user-images.githubusercontent.com/42869436/222512378-d6b04fca-87fe-49c8-a472-86c9b70353e7.png">

I am NOT able to successfully claim the hypercert because of the issue with test case 4.

[stop test]

**Comment by ryscheng** [comment]
@ccerv1 Can we close this out? 
Are the discovered bugs sufficiently captured by #414 and #420 ?

**Comment by holkeb** [comment]
TLDR
- All behavior as expected, except
    - I can mint and claim when I’m connected to Ethereum Mainnet, nothing happens, but I pay gas. I guess this is only related to Görli and Mainnet? 
    - I don’t see the “created” and “claimed” tag together
    - I see the the created hypercert twice, once to claim, once as created
    - Cosmetics: If unclaimed is 0.00%, we don't need to show it


Test 1
- I accidentally didn’t mint the hypercert on Görli, but was connected to Eth mainnet. It still created the hypercert (at least it showed the tooltip that it was successfully minted), but it doesn’t show up in my dashboard.
- All behavior as expected, when I minted it on Görli, except that it only has the tag “Claimed”, the tag “Created” is missing:
<img width="1098" alt="Pasted Graphic" src="https://user-images.githubusercontent.com/103380539/228073972-b84b4e7d-4258-4de5-a285-dedbfd4ef495.png">

- Cosmetics: If “Unclaimed” is 0.00%, we don’t need to show it
<img width="694" alt="Pasted Graphic 1" src="https://user-images.githubusercontent.com/103380539/228074138-5779db46-7bf1-433e-82f7-4838778fe9a5.png">

Test 2
- All behavior as expected, except
    - I saw the hypercert twice in the dashboard, once as claimable, once as created (the behaviour is fine, just cosmetics for some time later)
    - as in test 1 I only see the “claimed” tag (I will not refer to this in the following tests)
    - after claiming I saw that 7/8 hypercerts are claimed, even though there were only 7 hypercerts in my list. After opening some random hypercert and going back to the dashboard, it correctly shows 7/7

<img width="1115" alt="Pasted Graphic 3" src="https://user-images.githubusercontent.com/103380539/228074172-51a477ab-8990-4c17-b461-3194bd051bb9.png">


Test 3
- All behavior as expected, except
    - Same wrong temporary stats as in test 2: It shows 1/2 claimable hypercerts, even though only 1 is in the list. As above, it goes away after opening a hypercert and returning to the dashboard. (I will not refer to this behavior in the following test anymore)
- I played around with the networks again: I was able to claim it while being connected to Eth mainnet, it seemed as if it was successfully claimed, but nothing showed up. I was able to claim it later on Görli. Is this just an issue between Eth Mainnet and Görli? Or could this problem also arise between Eth Mainnet and Optimism? Not a big problem, but I paid gas fees for it :D

Test 4
- All behavior as expected.
- I did NOT experience the bug that Carl reported above.

Test 5
- All behavior as expected.

**Comment by holkeb** [comment]
I created an issue for the network bug as it also happens between Optimism and Eth Mainnet: https://github.com/hypercerts-org/hypercerts/issues/474

**Comment by ccerv1** [comment]
Great! Looks like most of the bugs are fixed. Do you want to create issues for:

- P1: A user should not be able to perform any contract interactions if they are connected to an unsupported network
- P2: A hypercert item should be able to display both the “created” and “claimed” tags together (if applicable to that cert)
- P2: Remove the "unclaimed" entry on the view page if it is equal to 0

@holkeb
